src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/ContigAlignmentsModifier.java

```diff
@@ -0,0 +1,208 @@
+                if (one.forwardStrand != two.forwardStrand) { // so that the inverted duplicated reference span is minimal.
+                    // jumpStart is for "the starting reference location of a jump that linked two alignment intervals", and
+                    // jumpLandingRefLoc is for "that jump's landing reference location"
+                    final int jumpStartRefLoc = one.referenceSpan.getEnd(),
+                            jumpLandingRefLoc = two.referenceSpan.getStart();
+                    oneYieldToTwo = jumpStartRefLoc <= jumpLandingRefLoc == one.forwardStrand;
+                } else {
+                    oneYieldToTwo = one.forwardStrand;
+                }
+            } else {
+                oneYieldToTwo = IntervalUtils.compareContigs(one.referenceSpan, two.referenceSpan, dictionary) > 0;
+            }
+
+            if (oneYieldToTwo) {
+                reconstructedOne = clipAlignmentInterval(one, overlapOnRead, true);
+                reconstructedTwo = two;
+            } else {
+                reconstructedOne = one;
+                reconstructedTwo = clipAlignmentInterval(two, overlapOnRead, false);
+            }
+
+        }
+        return Arrays.asList(reconstructedOne, reconstructedTwo);
+    }
+
+    /**
+     * Given {@code clipLengthOnRead} to be clipped from an aligned contig's {@link AlignmentInterval} {@code input},
+     * return a modified {@link AlignmentInterval} with the requested number of bases clipped away and ref span recomputed.
+     * @param input                 alignment to be modified (record not modified in place)
+     * @param clipLengthOnRead      number of read bases to be clipped away
+     * @param clipFrom3PrimeEnd     to clip from the 3' end of the read or not
+     */
+    private static AlignmentInterval clipAlignmentInterval(final AlignmentInterval input, final int clipLengthOnRead,
+                                                           final boolean clipFrom3PrimeEnd) {
+        Utils.validateArg(clipLengthOnRead < input.endInAssembledContig - input.startInAssembledContig + 1,
+                            "input alignment to be clipped away: " + input.toPackedString() + "\twith clip length: " + clipLengthOnRead);
+
+        final Tuple2<SimpleInterval, Cigar> result = computeNewRefSpanAndCigar(input, clipLengthOnRead, clipFrom3PrimeEnd);
+        final int newTigStart, newTigEnd;
+        if (clipFrom3PrimeEnd) {
+            newTigStart = input.startInAssembledContig;
+            newTigEnd   = input.endInAssembledContig - clipLengthOnRead;
+        } else {
+            newTigStart = input.startInAssembledContig + clipLengthOnRead;
+            newTigEnd   = input.endInAssembledContig;
+        }
+        return new AlignmentInterval(result._1, newTigStart, newTigEnd, result._2, input.forwardStrand, input.mapQual,
+                StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection.MISSING_NM,
```

Comment:
What's the difference between this constant and `AlignmentInterval.NO_NM`?

Reply:
removed and replaced with `NO_NM` everywhere